src/OpenTelemetry/Metrics/CompositeMetricReaderExt.cs

```diff
@@ -30,13 +32,16 @@ internal List<Metric> AddMetricsWithNoViews(Instrument instrument)
         for (var cur = this.Head; cur != null; cur = cur.Next)
         {
             var metric = cur.Value.AddMetricWithNoViews(instrument);
-            metrics.Add(metric);
+            if (metric != null)
```

Comment:
Why this change? Does it not affect the users of this list?

Reply:
I think it is a (rare) bug fix. This code is returning a list of all metrics published by the inner readers. Before it just assumed that each inner reader would return someting, but the inner reader may return `null` from `AddMetricWithNoViews`:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/c97a233a9a9d9656b8c431f69f5697f90554b3bb/src/OpenTelemetry/Metrics/MetricReaderExt.cs#L68

https://github.com/open-telemetry/opentelemetry-dotnet/blob/c97a233a9a9d9656b8c431f69f5697f90554b3bb/src/OpenTelemetry/Metrics/MetricReaderExt.cs#L84

Reply:
I think this accounted for in the method that consumes this list. If it's not required for enabling `nullable`, the I would strongly suggest we remove this change from this PR. There are already a lot changes in this PR. If this is indeed a bug that we need to fix, let's create a dedicated PR for it with the required unit tests added.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/c97a233a9a9d9656b8c431f69f5697f90554b3bb/src/OpenTelemetry/Metrics/CompositeMetricReaderExt.cs#L46-L49

Reply:
Interesting! OK I updated it so `null` is valid in the list. We should probably switch this to drop the `null`s and not add them to the list so that during measurement we don't need to waste cycles checking invalid indexes but fine to do as a follow-up.